### PR TITLE
Search: remove masterbar search menu to Calypso search admin page

### PR DESCRIPTION
**Pull request by kangzj**
#### Changes proposed in this Pull Request:
Remove search submenu from Calypso masterbar to allow (WP-Admin ) Search Dashbaord submenu to be registered. The PR needs to be tested on WPCOM.

We'll (1) merge the PR first and then (2) tweak `D84599-code` to allow Search Dashboard submenu to be added.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
pcNPJE-UW-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Ensure all GitHub checks pass
* Apply `D84599-code` to your WPCOM sandbox
* Sandbox `public-api.wordpress.com` and a simple site of your choice
* Ensure Search menu is removed in sidebar

|Before      | After    |
|----------|--------|
| image | image |

### CODE REVIEW

**Review by mmtr — COMMENTED** [review]
This will remove the Jetpack > Search menu item for self-hosted Jetpack sites too. Is that intentional?

Before | After
--- | ---
Screen Shot 2022-07-21 at 12 40 35 | Screen Shot 2022-07-21 at 12 41 11

I think we'll use D84235-code to add back the menu (but pointing now to WP Admin), but that will only work for Simple sites.

**Comment by kangzj** [comment]
Hi @mmtr, thanks for the feedback ❤️ 

> This will remove the Jetpack > Search menu item for self-hosted Jetpack sites too. Is that intentional?

Good catch! That is definitely not intended. I'll have another look. Thanks 

> I think we'll use D84235-code to add back the menu (but pointing now to WP Admin), but that will only work for Simple sites.

Search package will register submenu on atomic, so it should be okay. But still I'll need to check self hosted Jetpack sites.

**Comment by kangzj** [comment]
> This will remove the Jetpack > Search menu item for self-hosted Jetpack sites too. Is that intentional?

This is now fixed - changed the menu to be hidden only for simple sites just like Atomic ones 😄

**Review by mmtr — COMMENTED** [review]
Thanks! It works well now. I have a suggestion though – let me know what do you think.

Since we only need the "Jetpack > Search" menu in self-hosted Jetpack sites, what if we move the registration from the generic class here

https://github.com/Automattic/jetpack/blob/45cffc2b284e6f99f95a3a34504017f1afb220d3/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L389

to the specific class for Jetpack sites here

https://github.com/Automattic/jetpack/blob/45cffc2b284e6f99f95a3a34504017f1afb220d3/projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php#L138

That way we can get rid of the `hide_search_menu_for_calypso` function, and the resulting code would simpler.

**Comment by kangzj** [comment]
Hi @mmtr,

> That way we can get rid of the hide_search_menu_for_calypso function, and the resulting code would simpler.

Good point 👍

That makes it simpler and easier to understand. Addressed in 8a34b73. Thanks 😄 

Cheers

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php [hunk 1]
```diff
@@ -386,7 +386,6 @@ public function add_jetpack_menu() {
 		add_submenu_page( 'jetpack', esc_attr__( 'Activity Log', 'jetpack' ), __( 'Activity Log', 'jetpack' ), 'manage_options', 'https://wordpress.com/activity-log/' . $this->domain, null, 2 );
 		add_submenu_page( 'jetpack', esc_attr__( 'Backup', 'jetpack' ), __( 'Backup', 'jetpack' ), 'manage_options', 'https://wordpress.com/backup/' . $this->domain, null, 3 );
 		/* translators: Jetpack sidebar menu item. */
```
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: This comment should be moved too.

**Review by mmtr — COMMENTED** [review]
Nice, much simpler now! There is one super minor thing to address (but required to keep translations working). After solving that, this should be good to go.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php (hunk 1 above)
Comment (kangzj) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: @mmtr done in 8a34b73 😄

**Review by mmtr — APPROVED** [review]
:shipit: !

**Comment by kangzj** [comment]
Thanks @mmtr really appreciate your help 👍

**Pull request merged**